multiple definition of `HDMI_task'  (breaks firmware building)

I've tried to build '2.01' firmware without success.  Ia'm on latest master, haven't bisected yet (not sure if it's really needed, because it's probably obvious bug for a porgrammer)
66bfb400e8e8e343ec447a4deca307be077ec0f2 (HEAD -> master, origin/master, origin/HEAD)

```
make hdk2.hex 
cat gitdefines.h.tmp | tr "+=" "\#\042" > gitdefines.h
[HDK_20][-O3] ../src/Revision.c : HDK_20/src/Revision.o
[HDK_20] Linking HDK_20/HMD_MCU.elf
/usr/bin/avr-ld: HDK_20/src/main.o: in function `main':
/home/badpixel/Downloads/VR_HMD_VirtualRealityHeadset/Razer_Sensics_OSVR_HDK2/githubs/osvr_github/OSVR-HDK-MCU-Firmware/Source code/Embedded/makefile-build/../src/main.c:93: multiple definition of `HDMI_task'; HDK_20/src/SerialStateMachine.o:/home/badpixel/Downloads/VR_HMD_VirtualRealityHeadset/Razer_Sensics_OSVR_HDK2/githubs/osvr_github/OSVR-HDK-MCU-Firmware/Source code/Embedded/makefile-build/../src/SerialStateMachine.c:516: first defined here
/usr/bin/avr-ld: disabling relaxation; it will not work with multiple definitions
/usr/bin/avr-ld: HDK_20/src/main.o: in function `main':
/home/badpixel/Downloads/VR_HMD_VirtualRealityHeadset/Razer_Sensics_OSVR_HDK2/githubs/osvr_github/OSVR-HDK-MCU-Firmware/Source code/Embedded/makefile-build/../src/main.c:93: multiple definition of `HDMISwitch_task'; HDK_20/src/SerialStateMachine.o:/home/badpixel/Downloads/VR_HMD_VirtualRealityHeadset/Razer_Sensics_OSVR_HDK2/githubs/osvr_github/OSVR-HDK-MCU-Firmware/Source code/Embedded/makefile-build/../src/SerialStateMachine.c:516: first defined here
/usr/bin/avr-ld: HDK_20/src/uart_xmega.o:(.bss.HDMISwitch_task+0x0): multiple definition of `HDMISwitch_task'; HDK_20/src/SerialStateMachine.o:/home/badpixel/Downloads/VR_HMD_VirtualRealityHeadset/Razer_Sensics_OSVR_HDK2/githubs/osvr_github/OSVR-HDK-MCU-Firmware/Source code/Embedded/makefile-build/../src/SerialStateMachine.c:516: first defined here
/usr/bin/avr-ld: HDK_20/src/uart_xmega.o:(.bss.HDMI_task+0x0): multiple definition of `HDMI_task'; HDK_20/src/SerialStateMachine.o:/home/badpixel/Downloads/VR_HMD_VirtualRealityHeadset/Razer_Sensics_OSVR_HDK2/githubs/osvr_github/OSVR-HDK-MCU-Firmware/Source code/Embedded/makefile-build/../src/SerialStateMachine.c:516: first defined here
/usr/bin/avr-ld: HDK_20/src/DeviceDrivers/VideoInput.o: in function `VideoInput_Get_Status':
/home/badpixel/Downloads/VR_HMD_VirtualRealityHeadset/Razer_Sensics_OSVR_HDK2/githubs/osvr_github/OSVR-HDK-MCU-Firmware/Source code/Embedded/makefile-build/../src/DeviceDrivers/VideoInput.c:94: multiple definition of `HDMI_task'; HDK_20/src/SerialStateMachine.o:/home/badpixel/Downloads/VR_HMD_VirtualRealityHeadset/Razer_Sensics_OSVR_HDK2/githubs/osvr_github/OSVR-HDK-MCU-Firmware/Source code/Embedded/makefile-build/../src/SerialStateMachine.c:516: first defined here
/usr/bin/avr-ld: HDK_20/src/DeviceDrivers/VideoInput.o: in function `VideoInput_Get_Status':
/home/badpixel/Downloads/VR_HMD_VirtualRealityHeadset/Razer_Sensics_OSVR_HDK2/githubs/osvr_github/OSVR-HDK-MCU-Firmware/Source code/Embedded/makefile-build/../src/DeviceDrivers/VideoInput.c:94: multiple definition of `HDMISwitch_task'; HDK_20/src/SerialStateMachine.o:/home/badpixel/Downloads/VR_HMD_VirtualRealityHeadset/Razer_Sensics_OSVR_HDK2/githubs/osvr_github/OSVR-HDK-MCU-Firmware/Source code/Embedded/makefile-build/../src/SerialStateMachine.c:516: first defined here
collect2: error: ld returned 1 exit status
make: *** [add_variant.mk:88: HDK_20/HMD_MCU.elf] Error 1
```

git status
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean
 badpixel \ … \ Source code \ Embedded \ makefile-build \ make hdk2_svr
make: *** No rule to make target 'hdk2_svr'.  Stop.

 - -

#make all
cat gitdefines.h.tmp | tr "+=" "\#\042" > gitdefines.h
[HDK_20][-O3] ../src/Revision.c : HDK_20/src/Revision.o
[HDK_20] Linking HDK_20/HMD_MCU.elf
/usr/bin/avr-ld: HDK_20/src/main.o: in function `main':
/home/badpixel/Downloads/VR_HMD_VirtualRealityHeadset/Razer_Sensics_OSVR_HDK2/githubs/osvr_github/OSVR-HDK-MCU-Firmware/Source
code/Embedded/makefile-build/../src/main.c:93: multiple definition of
`HDMI_task'; HDK_20/src/SerialStateMachine.o:/home/badpixel/Downloads/VR_HMD_VirtualRealityHeadset/Razer_Sensics_OSVR_HDK2/githubs/osvr_github/OSVR-HDK-MCU-Firmware/Source
code/Embedded/makefile-build/../src/SerialStateMachine.c:516: first
defined here
/usr/bin/avr-ld: disabling relaxation; it will not work with multiple
definitions
/usr/bin/avr-ld: HDK_20/src/main.o: in function `main':
/home/badpixel/Downloads/VR_HMD_VirtualRealityHeadset/Razer_Sensics_OSVR_HDK2/githubs/osvr_github/OSVR-HDK-MCU-Firmware/Source
code/Embedded/makefile-build/../src/main.c:93: multiple definition of
`HDMISwitch_task';
HDK_20/src/SerialStateMachine.o:/home/badpixel/Downloads/VR_HMD_VirtualRealityHeadset/Razer_Sensics_OSVR_HDK2/githubs/osvr_github/OSVR-HDK-MCU-Firmware/Source
code/Embedded/makefile-build/../src/SerialStateMachine.c:516: first
defined here
/usr/bin/avr-ld: HDK_20/src/uart_xmega.o:(.bss.HDMISwitch_task+0x0):
multiple definition of `HDMISwitch_task';
HDK_20/src/SerialStateMachine.o:/home/badpixel/Downloads/VR_HMD_VirtualRealityHeadset/Razer_Sensics_OSVR_HDK2/githubs/osvr_github/OSVR-HDK-MCU-Firmware/Source
code/Embedded/makefile-build/../src/SerialStateMachine.c:516: first
defined here
/usr/bin/avr-ld: HDK_20/src/uart_xmega.o:(.bss.HDMI_task+0x0):
multiple definition of `HDMI_task';
HDK_20/src/SerialStateMachine.o:/home/badpixel/Downloads/VR_HMD_VirtualRealityHeadset/Razer_Sensics_OSVR_HDK2/githubs/osvr_github/OSVR-HDK-MCU-Firmware/Source
code/Embedded/makefile-build/../src/SerialStateMachine.c:516: first
defined here
/usr/bin/avr-ld: HDK_20/src/DeviceDrivers/VideoInput.o: in function
`VideoInput_Get_Status':
/home/badpixel/Downloads/VR_HMD_VirtualRealityHeadset/Razer_Sensics_OSVR_HDK2/githubs/osvr_github/OSVR-HDK-MCU-Firmware/Source
code/Embedded/makefile-build/../src/DeviceDrivers/VideoInput.c:94:
multiple definition of `HDMI_task';
HDK_20/src/SerialStateMachine.o:/home/badpixel/Downloads/VR_HMD_VirtualRealityHeadset/Razer_Sensics_OSVR_HDK2/githubs/osvr_github/OSVR-HDK-MCU-Firmware/Source
code/Embedded/makefile-build/../src/SerialStateMachine.c:516: first
defined here
/usr/bin/avr-ld: HDK_20/src/DeviceDrivers/VideoInput.o: in function
`VideoInput_Get_Status':
/home/badpixel/Downloads/VR_HMD_VirtualRealityHeadset/Razer_Sensics_OSVR_HDK2/githubs/osvr_github/OSVR-HDK-MCU-Firmware/Source
code/Embedded/makefile-build/../src/DeviceDrivers/VideoInput.c:94:
multiple definition of `HDMISwitch_task';
HDK_20/src/SerialStateMachine.o:/home/badpixel/Downloads/VR_HMD_VirtualRealityHeadset/Razer_Sensics_OSVR_HDK2/githubs/osvr_github/OSVR-HDK-MCU-Firmware/Source
code/Embedded/makefile-build/../src/SerialStateMachine.c:516: first
defined here
collect2: error: ld returned 1 exit status
make: *** [add_variant.mk:88: HDK_20/HMD_MCU.elf] Error 1


On 9/28/20, Ryan A. Pavlik <notifications@github.com> wrote:
> Hmm, I built this recently and didn't see any issues. the make command you
> want is `make hdk2_svr`
>
> --
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/OSVR/OSVR-HDK-MCU-Firmware/issues/36#issuecomment-700155028

--- a/Source code/Embedded/src/main.h
+++ b/Source code/Embedded/src/main.h
@@ -47,7 +47,7 @@
 #include "usb_protocol_cdc.h"
 
 // status of the various periodic tasks
-bool HDMI_task;
-bool HDMISwitch_task;
+extern bool HDMI_task;
+extern bool HDMISwitch_task;
 
 #endif  // _MAIN_H_

